### ISSUE TITLE
How to dynamically split tensor

### ISSUE THREAD

**Issue by RunningLeon** [pytorch/vision#2994]
##  How to split a tensor dynamically by split_sizes, not by constant shape
Trying to convert mask-rcnn to onnx and run on onnxruntime.
Following code try to split mask_pred by num_mask_roi_per_img
However, while run in onnxruntime, num_mask_roi_per_img becomes constant value, for instance (68,) which is number of boxes while tracing.

```
# split batch mask prediction back to each image
num_mask_roi_per_img = [ det_bbox.shape[0] for det_bbox in det_bboxes ]
mask_preds = mask_pred.split(num_mask_roi_per_img, 0)
```
It got this error while run in onnxruntime with another input image.

> 

<class 'onnxruntime.capi.onnxruntime_pybind11_state.InvalidArgument'>", "[ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Non-zero status code returned while running SplitToSequence node. Name:'SplitToSequence_1396' Status Message: split_size_sum (68) != split_dim_size (23)"

Could anyone help me with this?
Many thanks in advance.

cc @neginraoof

**Comment by zhiqwang** [comment]
Hi @RunningLeon , It's difficult to reproduce your bugs for the information you have supplied.

TorchVision's have similar ops when exporting the object detection models:

https://github.com/pytorch/vision/blob/80f41f8d32b1fcb380d5df2116063af7034ff29a/torchvision/models/detection/roi_heads.py#L77-L81

**Comment by fmassa** [comment]
@neginraoof could you have a look?

**Comment by RunningLeon** [comment]
@zhiqwang Thanks. I would provide a sample code if possible.

**Comment by RunningLeon** [comment]
With sample code, the error cannot be reproduced. Will reopen it if necessary. 
```
import os
import torch
import onnx
import numpy as np
import onnxruntime as rt

class DummyModel(torch.nn.Module):
    
    def forward(self, bboxes, labels, num_class: int=3):
        """
        try to reorganize bboxes by labels, concat and then test split
        """
        bbox_li = []
        for i in range(num_class):
            inds = labels.eq(i).nonzero(as_tuple=False).flatten()
            bbox_li.append(bboxes[inds])
        num_li = [b.size(0) for b in bbox_li]
        cat = torch.cat(bbox_li, dim=0)
        result = cat.split(num_li, dim=0)
        return result
    
num = 5
num_class = 3
boxes = torch.randint(10, size=(num, 4), dtype=torch.float32)
labels = torch.randint(num_class, size=(num,), dtype=torch.long)
input_data = (boxes, labels)
model = DummyModel()
onnx_io = "test_split.onnx"

input_names = ['boxes', 'labels']
output_names = None
torch.onnx.export(
    model,
    input_data,
    onnx_io,
    verbose=False,
    input_names=input_names,
    output_names=output_names,
    opset_version=11
)

boxes = torch.randint(10, size=(num, 4), dtype=torch.float32)
# labels = torch.randint(num_class, size=(num,), dtype=torch.long) # not support dynamic type
print(f'labels={labels}')
input_data = (boxes, labels)
desired = model(*input_data)
print(f'Expect:\n{desired}')

sess = rt.InferenceSession(onnx_io)
input_names = [_.name for _ in sess.get_inputs()]
output_names = [_.name for _ in sess.get_outputs()]

print(f'input_names={input_names}')
print(f'output_names={output_names}')

#forward model
input_data_npy = [i.detach().cpu().numpy() for i in input_data]
actual = sess.run(None, {name : data for name, data in zip(input_names, input_data_npy)})
actual = actual
print(f'Actual:\n{actual}')
print(f'Actual:\n{actual}')
for d, a in zip(desired, actual):
    np.testing.assert_allclose(d, a, rtol=1e-05, atol=1e-05)
print('All same.')
```
Thanks.

**Comment by TriLoo** [comment]
Do you remember how to solve this problem finally. 
I meet same error but cannot figure out how to solve it. 

thanks.
